From 77f265e1c59a4f7e92b9e1f50e76214e1da694c3 Mon Sep 17 00:00:00 2001
From: Alexander Sack <asac@pantacor.com>
Date: Tue, 7 Feb 2023 09:08:11 +0100
Subject: [PATCH 307/310] fix create_and_bind_to_netlink setting sa.nl_pid to
 potentially not system unique value

(cherry picked from commit 5b34ff9e9804b11f851c2c0e6782e015de6e790f)
---
 libbb/xconnect.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libbb/xconnect.c b/libbb/xconnect.c
index 095f7ff..e99c59b 100644
--- a/libbb/xconnect.c
+++ b/libbb/xconnect.c
@@ -414,7 +414,7 @@ int FAST_FUNC create_and_bind_to_netlink(int proto, int grp, unsigned rcvbuf)
 
 	memset(&sa, 0, sizeof(sa));
 	sa.nl_family = AF_NETLINK;
-	sa.nl_pid = getpid();
+	sa.nl_pid = 0;
 	sa.nl_groups = grp;
 	xbind(fd, (struct sockaddr *) &sa, sizeof(sa));
 	close_on_exec_on(fd);
-- 
2.43.0

